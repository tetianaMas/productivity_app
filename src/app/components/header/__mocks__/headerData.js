const settingsData = {
  buttonsValues: {
    iteration: { default: 5, max: 5, min: 2, step: 1, value: 5 },
    longBreak: { default: 30, max: 30, min: 15, step: 5, value: 30 },
    shortBreak: { default: 5, max: 5, min: 3, step: 1, value: 5 },
    work: { default: 25, max: 25, min: 15, step: 5, value: 25 },
  },
  cycleCount: 2,
  firstCycleTime: {
    full: 175,
    hours: 2,
    minutes: 55,
  },
  fullCycleTime: {
    full: 320,
    hours: 5,
    minutes: 20,
  },
  minutesInSchedule: 30,
  secondCycleTime: 145,
  timeScheduleDelimeter: 20,
};

export const context = {
  tab: {
    firstVal: 'pomodoros',
    secondVal: 'categories',
    activeTab: true,
    firstValAttr: 'data-tab="pomodoros"',
    secondValAttr: 'data-tab="categories"',
  },
  settingsData,
};
