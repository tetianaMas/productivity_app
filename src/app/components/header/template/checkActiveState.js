/**
 * @description manages page hash and active css class according to it
 * @exports checkActiveState
 * @param {string} btnValue
 * @return {string} name of css class which represents active state of navigation button
 */
export default function (btnValue) {
  const pageHash = `${window.location.hash}/`.match(/^#\/(.*?)\//);

  if (pageHash && pageHash[1] === btnValue) {
    return 'navigation__btn--active';
  }
}
