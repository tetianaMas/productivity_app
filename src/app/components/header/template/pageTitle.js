/**
 * @description manages page title according to current route
 * @exports pageTitle
 * @return {string} name of page according to current route
 */
export default function () {
  const pageTitles = {
    default: 'daily task list',
    'task-list': 'daily task list',
    settings: 'settings',
    reports: 'report',
    timer: ' ',
  };
  const pageHash = `${window.location.hash}/`.match(/^#\/(.*?)\//);

  if (pageHash) {
    const pageTitle = pageTitles[pageHash[1]];

    if (pageTitle) {
      return pageTitle;
    }
  }

  return pageTitles.default;
}
