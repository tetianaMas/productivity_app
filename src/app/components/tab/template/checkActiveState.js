/**
 * @description manages tabs for reports
 * @exports checkActiveState
 * @param {boolean} period if reports by time should be shown
 * @param {string} value current tab value
 * @return {string} active css class
 */
export default function (period, value) {
  if (period) {
    const regex = /(^#\/reports\/)(.*)\//gm;
    const match = regex.exec(`${window.location.hash}`);
    let time;

    if (match) {
      time = match[2];
    }

    if (time === value) {
      return 'tab__btn--active';
    }
  } else {
    const regex = /(^#\/reports\/)(.*)\/(.*)$/gm;
    const match = regex.exec(`${window.location.hash}`);
    let category;

    if (match) {
      category = match[3];
    }

    if (category === value) {
      return 'tab__btn--active';
    }
  }
}
