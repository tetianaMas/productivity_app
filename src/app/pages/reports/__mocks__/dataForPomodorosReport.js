import { Utils } from '../../../helpers/utils';

export const dataForPomodorosReport = [
  {
    date: Utils.getCurrentDate(),
    failed: 7,
    high: 1,
    low: 0,
    middle: 0,
    urgent: 3,
  },
];
