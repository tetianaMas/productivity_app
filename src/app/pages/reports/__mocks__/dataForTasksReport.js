import { Utils } from '../../../helpers/utils';

export const dataForTasksReport = [
  {
    date: Utils.getCurrentDate(),
    failed: 1,
    high: 0,
    low: 0,
    middle: 0,
    urgent: 1,
  },
];
