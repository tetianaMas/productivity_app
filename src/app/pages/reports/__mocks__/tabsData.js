export const tabsData = {
  additionalClasses: 'tab--center js-tab-top',
  firstVal: 'day',
  firstValAttr: 'data-reports-period="day"',
  secondVal: 'week',
  secondValAttr: 'data-reports-period="week"',
  thirdVal: 'month',
  thirdValAttr: 'data-reports-period="month"',
  period: true,
  bottomTabs: {
    additionalClasses: 'tab--center tab--bottom js-tab-bottom',
    firstVal: 'pomodoros',
    firstValAttr: 'data-reports-category="pomodoros"',
    secondVal: 'tasks',
    secondValAttr: 'data-reports-category="tasks"',
    secondValActive: true,
  },
};
