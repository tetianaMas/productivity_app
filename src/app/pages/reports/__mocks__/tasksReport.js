import { Utils } from '../../../helpers/utils';

export const tasks = [
  {
    categoryId: 'hobby',
    completeDate: Utils.getCurrentDate(),
    completedCount: [0],
    createDate: { day: 17, fullDeadline: 1615978347385, month: 'March' },
    deadline: 'April 10, 2021',
    deadlineDate: { day: '', fullDeadline: 1616716800000, month: 'today' },
    description: 'Wow',
    estimation: '5',
    failedPomodoros: [0, 1, 2, 3, 4],
    id: '00807824925639886039',
    priority: 'high',
    status: 'COMPLETED',
    title: 'Wow',
  },
  {
    categoryId: 'hobby',
    completeDate: Utils.getCurrentDate(),
    completedCount: [2, 3, 4],
    createDate: { day: 2, fullDeadline: 1617314051232, month: 'April' },
    deadline: 'April 15, 2021',
    deadlineDate: { day: 15, fullDeadline: 1618434000000, month: 'today' },
    description: 'Cool',
    estimation: '3',
    failedPomodoros: [0, 1],
    id: '08835933769418409683',
    priority: 'urgent',
    status: 'COMPLETED',
    title: 'Cool',
  },
];
