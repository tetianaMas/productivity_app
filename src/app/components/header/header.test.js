/* eslint-disable no-undef */
import checkActiveState from './template/checkActiveState';
import pageTitle from './template/pageTitle';
import headerTemplate from './template/header.hbs';
import { context } from './__mocks__/headerData';

test('renders correctly', () => {
  const tree = JSON.stringify(headerTemplate(context));
  expect(tree).toMatchSnapshot();
});

describe('handlebars helpers', () => {
  let windowSpy;

  beforeEach(() => {
    windowSpy = jest.spyOn(window, 'window', 'get');
    windowSpy.mockImplementation(() => ({
      location: {
        hash: '#/settings',
      },
    }));
  });

  afterEach(() => {
    windowSpy.mockRestore();
  });

  test('should return active btn css class', () => {
    const result = checkActiveState('settings');
    const truthyResult = 'navigation__btn--active';

    expect(truthyResult).toEqual(result);
  });

  test('should be undefined.', () => {
    const result = checkActiveState('wrong data');

    expect(result).toBeUndefined();
  });

  test('should return the right title.', () => {
    const result = pageTitle();

    expect(result).toBe('settings');
  });
});
