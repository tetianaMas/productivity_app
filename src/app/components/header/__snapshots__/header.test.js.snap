// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"\\" <header class=\\\\\\"header js-header-main\\\\\\">\\\\r\\\\n\\\\r\\\\n <h1 class=\\\\\\"header__title\\\\\\">\\\\r\\\\n    daily task list\\\\r\\\\n  </h1>\\\\r\\\\n  <div class=\\\\\\"header__logo d-none\\\\\\">\\\\r\\\\n    <a href=\\\\\\"#\\\\\\">\\\\r\\\\n      <img src=\\\\\\"../images/Logo.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" />\\\\r\\\\n    </a>\\\\r\\\\n  </div>\\\\r\\\\n  <nav class=\\\\\\"navigation\\\\\\" aria-label=\\\\\\"Main navigation\\\\\\">\\\\r\\\\n    <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-list \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-statistics \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-settings \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n    </ul>\\\\r\\\\n  </nav>\\\\r\\\\n</header>\\""`;
